### PR TITLE
fix(exp and mxp and oob and ecdata): turned toBigInteger() into toUnsignedBigInteger

### LINKED ISSUE #762: unsignedBigInteger
Unless you're sure you want signed number, don't use `toBigInteger()` but  `.toUnsignedBigInteger()`

Seen in MXP, ECDATA, OOB

### PR DESCRIPTION
**Pull request by lorenzogentile404**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/exp/ExpOperation.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ final void traceComputation(int stamp, Trace trace) {
       pComputationMsb
       */
       // tanzb turns to 1 iff trimAcc is nonzero
-      tanzb = pComputationTrimAcc.slice(0, i + 1).toBigInteger().signum() != 0;
+      tanzb = pComputationTrimAcc.slice(0, i + 1).toUnsignedBigInteger().signum() != 0;
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/exp/ExpOperation.java:94]: it seems here you really want a signed bigInt as you ask for its sign no ?

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public OobOperation(
   private void setOpCodeFlagsAndWghtSumAndIncomingInst(MessageFrame frame) {
     final OpCode opCode = OpCode.of(frame.getCurrentOperation().getOpcode());
     // In the case of CALLs and CREATEs this value will be replaced
-    wghtSum = UnsignedByte.of(opCode.byteValue()).toBigInteger();
+    wghtSum = BigInteger.valueOf(Byte.toUnsignedInt(opCode.byteValue()));
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java:194]: sounds a complicate way of life to get it :)

**Review by letypequividelespoubelles — CHANGES_REQUESTED** [review]
it seems you need to keep a signed BigInt for exp ?

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/exp/ExpOperation.java (hunk 1 above)
Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/exp/ExpOperation.java:94]: No, you just want to distinguish between `0` and `!= 0`. The alternative would be `!...toUnsignedBigInteger().compareTo(BigInteger.ZERO)` or creating some dedicated method for it.

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java (hunk 2 above)
Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java:194]: Both of them seem to work and are comparable in terms of readability. Do you have suggestions for something better?

Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java:194]: at some point `wghtSum` is assigned to `oobInst`, that is expected to be a `bigInteger`, so even if I change the type of `wghtSum` to something more convenient, I need that conversion anyway later.

Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java:194]: Sorry, I am wrong. `oobInst` is now `Bytes`, so potentially we may convert both of them to `Bytes`.

Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java:194]: `wghtSum = Bytes.fromHexString("FF08").toUnsignedBigInteger();` due to this it cannot simply be a byte. I would suggest to leave it as it is.

**Review by letypequividelespoubelles — APPROVED** [review]
LGTM

**Pull request merged**